### ISSUE TITLE
Parsing JSON with `ALLOW_MISSING_VALUE` enabled results in endless stream of `VALUE_NULL` tokens

### ISSUE THREAD

**Issue by jusliu** [FasterXML/jackson-core#616]
```
import com.fasterxml.jackson.core.JsonFactory;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.JsonParser.Feature;
import com.fasterxml.jackson.core.JsonToken;
import java.io.IOException;

public class Test {
	public static void main(String[] args) {
		String content = ",";
		JsonFactory f = new JsonFactory().configure(Feature.ALLOW_MISSING_VALUES, true);
		try {
			JsonParser parser = f.createJsonParser(content);
			JsonToken current = parser.nextToken();

			// this will result in an infinite loop
			while (parser.hasCurrentToken()) {
				System.out.println(current); // prints VALUE_NULL
				current = parser.nextToken();
			}
		} catch (IOException e) {
			System.out.println(e);
		}
	}
}
```

This code results in an infinite loop. Only happens with ALLOW_MISSING_VALUES enabled. Treats the unexpected comma as a missing value or something, replacing it with NULL, but on the top-level this results in an infinite loop where nextToken() will return VALUE_NULL forever.

Also happens on inputs including:

1. [],
2. {"foo": "bar"},

**Comment by cowtowncoder** [comment]
Thank you for reporting this. Which version is this verified with?

**Comment by jusliu** [comment]
I was originally using 2.10.1, but I just tested this on 2.11.0 and it's still occurring.

**Comment by cowtowncoder** [comment]
Added failing test.

**Comment by cowtowncoder** [comment]
One quick note: commas should not really be accepted as separators in root level so just realizing that this needs to result in parsing error. But bit worried that someone somewhere might already be relying on such behavior...

**Comment by cowtowncoder** [comment]
Fixed: will now report parsing error for root-level commas, which will also prevent infinite sequences of `VALUE_NULL`s. Fix will be in 2.10.5.
